src/components/Table.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import FaSort = require('react-icons/lib/fa/sort');
+import styled from 'styled-components';
+
+import Button from './Button';
+import { Flex } from './Grid';
+import theme from '../theme';
+
+type TableDataItem =
+	| void
+	| string
+	| JSX.Element
+	| {
+			value: any;
+			display: string | JSX.Element;
+		};
+
+export interface TableDataRow {
+	[key: string]: TableDataItem;
+}
+
+type TableSortFunction = <T>(a: T, b: T) => number;
+
+interface TableHeaderItem {
+	field: string;
+	icon?: string;
+	label?: string | JSX.Element;
+	sortable?: boolean | TableSortFunction;
+}
+
+interface TableProps {
+	header: TableHeaderItem[];
+	data: TableDataRow[];
+}
+
+const BaseTable = styled.table`
+	width: 100%;
+	border-spacing: 0;
+
+	> thead {
+		background-color: #f2f2f2;
+
+		> tr > th {
+			padding-left: 40px;
+			padding-top: 10px;
+			padding-bottom: 10px;
+			font-size: 16px;
+		}
+	}
+
+	> tbody {
+		> tr {
+			> td {
+				font-size: 16px;
+				padding-top: 14px;
+				padding-bottom: 14px;
+				padding-left: 40px;
+			}
+
+			&:nth-of-type(even) {
+				background-color: #f8f8f8;
+			}
+		}
+	}
+`;
+
+const HeaderButton = styled(Button)`
+	display: block;
+`
+
+const renderField = (row: TableDataRow, field: string) => {
+	const value = get(row, field);
+
+	if (!value) {
+		return '';
+	}
+
+	return value.hasOwnProperty('display') ? (value as any).display : value;
+};
+
+export default class Table extends React.Component<
+	TableProps,
+	{
+		sortColumn: null | string;
+		reverse: boolean;
+	}
+> {
+	constructor(props: TableProps) {
+		super(props);
+
+		this.state = {
+			sortColumn: null,
+			reverse: false,
+		};
+	}
+
+	sortData(data: TableDataRow[]): TableDataRow[] {
+		if (this.state.sortColumn === null) {
+			return data;
+		}
+
+		const collection = (sortBy(data.slice(), item => {
+			const sortableValue = item[this.state.sortColumn!];
+			return isPlainObject(sortableValue)
+				? (sortableValue as any).value
+				: sortableValue;
+		}) as any) as TableDataRow[];
+
+		if (this.state.reverse) {
+			reverse(collection);
+		}
+
+		return collection;
+	}
+
+	toggleSort(field: string) {
+		if (this.state.sortColumn === field) {
+			this.setState({ reverse: !this.state.reverse });
+			return;
+		}
+		this.setState({
+			sortColumn: field,
+			reverse: false,
+		});
+	}
+
+	render() {
+		const { header, data, ...props } = this.props;
+
+		return (
+			<BaseTable {...props}>
+				<thead>
+					<tr>
+						{map(header, item => {
+							if (item.sortable) {
+								return (
+									<th key={item.field}>
+										<HeaderButton
+											plaintext
+											primary={this.state.sortColumn === item.field}
+											onClick={() =>
+												this.toggleSort(item.field)}
+										>
+											<Flex align="center">
+												{item.label || item.field}
+												&nbsp;
+												<FaSort color={this.state.sortColumn === item.field ? theme.colors.info.main : ''} />
```

Comment:
That's probably forgotten but I guess that the icons should change based on the current sorting state.

Reply:
Interestingly, this is how the current sort headers work and how the zeplin designs show it. Lets talk to @konmouz and see if he has any opinions